pallets/msa/src/lib.rs

```diff
@@ -173,10 +159,20 @@ pub mod pallet {
 		type ProposalProvider: ProposalProvider<Self::AccountId, Self::Proposal>;
 	}
 
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
 	#[pallet::pallet]
 	#[pallet::generate_store(pub (super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
 	pub struct Pallet<T>(_);
 
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
+			migration::migrate::<T>()
+		}
+	}
+
```

Comment:
you might not want to do this in pallet hooks,  instead use `OnRuntimeUpgrade` in runtime
https://github.com/LibertyDSNP/frequency/commit/f05867c0cafb0f0bbc6f1dd4a6c5eceab8db536c

Reply:
It was hard to tell which was better, but this way appeared to be the latest pattern? Thoughts as to why one way instead of another?

Reply:
OnRuntimeUpgrade is preferred

Parity states a warning for using hook

https://paritytech.github.io/substrate/master/frame_support/traits/trait.Hooks.html#method.on_runtime_upgrade

> While this function can be freely implemented, using on_runtime_upgrade from inside the pallet is discouraged and might get deprecated in the future. Alternatively, export the same logic as a free-function from your pallet, and pass it to type Executive from the top-level runtime.
> 
> [source](https://paritytech.github.io/substrate/master/src/frame_support/traits/hooks.rs.html#275-277)
> 
>

Reply:
Updated... I think